org.openhab.binding.zigbee/src/test/java/org/openhab/binding/zigbee/internal/ZigBeeDeviceConfigHandlerTest.java

```diff
@@ -0,0 +1,66 @@
+package org.openhab.binding.zigbee.internal;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+
+import com.zsmartsystems.zigbee.ZigBeeEndpoint;
+import com.zsmartsystems.zigbee.ZigBeeNode;
+import com.zsmartsystems.zigbee.zcl.ZclCluster;
+import com.zsmartsystems.zigbee.zcl.protocol.ZclDataType;
+
+/**
+ *
+ * @author Chris Jackson - Initial contribution
+ *
+ */
+public class ZigBeeDeviceConfigHandlerTest {
+
+    @Ignore
```

Comment:
Currently I have problems running tests requiring Mockito - I'm hoping that this will be resolved soon as I know there are moves afoot to fix the dependancy issues.

Reply:
Here one could use `Mockito.anyInt()` (instead of `ArgumentMatchers.anyInt()`) to make the test run with Mockito (see my other comment for the other test on this for few more details).

Reply:
For whatever reason, as soon as I save within the IDE, Eclipse changes ```Mockito.anyInt()``` back to ```ArgumentMatchers.anyInt()```. Do you have any idea why Eclipse does is?

It does resolve the issue if I edit it outside of Eclipse, but it prevents me using tests in Eclipse.

Reply:
You probably have automatic cleanup on save enabled in Eclipse, and then the rule 'Change indirect accesses to static members to direct accesses (accesses through subtypes)' is applied by Eclipse.

As Eclipse only applies this cleanup rule if the static method is actually defined in the parent class, I looked at the used mockito versions, which leads to the actual problem (I believe):
* Eclipse handles the project as a plugin project, and, hence, retrieves dependencies needed to satisfy the imports from the bundle manifest from the target platform. The mockito version in the openhab target platform is 2.x.
* Maven uses surefire to execute the tests, and, hence the dependencies specified in the pom.xml - and there mockito is imported in versino 1.x.
* In Mockito 2.x, the argument matchers are defined in the `ArgumentMatchers` class; in Mockito 1.x, the argument matchers are defined in Mockito's own `Matcher` class.

That is why maven requires to use `Mockito.anyInt()`, while Eclipse wants to change this to `ArgumentMatchers.anyInt()`.

For me, this raises the question how the project setup should look like (regarding maven and eclipse configuration). Should the tests by run by tycho-surefire instead of surefire, so that maven uses the same dependencies as eclipse does now, or should the eclipse project be configured differently? But this is probably beyond the scope of this PR :)

Reply:
As described in my previous comment, the issue comes from different Mockito versions. I just saw that maven surefire uses both the dependencies declared in the maven `<dependencies>` section and the p2 dependencies derived from the manifest.

In consequence, I think that all dependencies (except the dependency to junit itself, which is needed by maven surefire) should be removed from the pom.xml `<dependencies>` section, so that there is only a single source for dependencies used in the test (viz the manifest). This would mean removing the dependency to mockito-all and to slf4j-simple from the pom.xml.

What do you think?